phac-nml/irida-next
# Namespace group link UI create

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI functionality to create a namespace group link for both groups and projects.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/3aaf78ba-94ad-422c-bba3-64e6387a4dcf)

![image](https://github.com/phac-nml/irida-next/assets/25466211/456d2fa6-bb53-4826-8d1a-7548662e12f6)

![image](https://github.com/phac-nml/irida-next/assets/25466211/0032fa10-6c15-4db2-9a10-362345095480)

![image](https://github.com/phac-nml/irida-next/assets/25466211/88647656-5222-4cca-a747-ceafbe60f825)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as any user that has groups and projects
2. Go the the members page for a group
3. You should see an `Invite group` button and 2 tabs (Members, Groups)
4. Click the `Invite group` button and select a group to share the namespace with, the group access level, and optionally select an expiration date (currently no restrictions)
5. Click the `Share` button
6. Click the `Groups` tab if you have not already done so
7. The new namespace group link should be listed. Note by default 5 namespace group links are seeded for each group and project. Ensure that the source is direct member for this new namespace group link and that the access level and optional expiration are also listed correctly

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
